fondant/manifest.py

```diff
@@ -249,6 +249,12 @@ def evolve(  # noqa : PLR0912 (too many branches)
                     evolved_manifest.subsets[subset_name].add_field(
                         field.name, field.type, overwrite=True
                     )
+
+                # Update subset location as this is currently always rewritten
+                evolved_manifest.subsets[subset_name]._specification[
+                    "location"
+                ] = f"/{subset_name}/{self.run_id}/{component_id}"
```

Comment:
When trying it out locally, the location does seem to be updated:
```
{"metadata": {"base_path": "/Users/nielsrogge/Documents/fondant_artifacts", "run_id": "test_niels", "component_id": "download_images"}, 

"index": {"location": "/index/test_niels/download_images"}, 

"subsets": {"images": {"location": "/images/test_niels/download_images", "fields": {"url": {"type": "utf8"}, "data": {"type": "binary"}, "width": {"type": "int16"}, "height": {"type": "int16"}}}}}
```

Reply:
> Moreover, the manifest says that the "images" subset includes a field "url" but the actual data only contains "data", "width" and "height" columns as there's no coupling between the evolve method and the actual dataframe. This is because the previous component (laion_retrieval) added a URL field, which the image downloader doesn't return back.

Specifying `additionalFields: false` in the images output subset in the component spec should fix this. We still need functionality to match the produced data against the manifest.

I don't see why the location would only work locally. Maybe the version running on kfp wasn't the one containing your fix?

Reply:
Ok update, location wasn't updated due to the component not using the updated version of Fondant. Have included `additionalFields: false` as well but will create a ticket to add functionality to check the produced data against the manifest.